Feature/query refactoring and validations 2395

tilt opened the pull request:
https://www.openproject.org/work_packages/2395

nkenneweg committed:
improves objects to be the same for queryWHierarchy

ulferts committed:
no longer serializes changes as they are declared to be serialized

As of rails 2.3.17, assigning a serialized object to an AR attribute declared as serialized is not permitted

GarstgerUnhold committed:
small fixes for unified project objects

nkenneweg committed:
Merge pull request #42 from opf/fix/project_hierarchy_same_objects

improves objects to be the same for queryWHierarchy

nkenneweg committed:
this fixes the hierarchy for searches.

Commit:
bug fix: Changing the role of a project member/group should not remove it from the project.

- Changing the role of a project member should not trigger a deletion.
- The project member should however be destroyed as soon as all inherited roles have been removed and it has no (direct) roles assigned to it.
- Test have been added to project core_tests.

nkenneweg committed:
bugfix merge for double selection.

nkenneweg committed:
improved addUnmatchedParents to also reflect selection.

nkenneweg committed:
improved adding of selected/unmatched parents.
improved data format.
if new data is loaded and we are currently down a project tree,
the parents of the previous element are taken into consideration.

nkenneweg committed:
improved even more. Previous solution was buggy-
also removed dead code.

nkenneweg committed:
removed bug with select2 scroll point.

apfelfabrik committed:
Merge pull request #45 from opf/fix/member_role_change

bug fix: Changing the role of a project member/group.

nkenneweg committed:
bugfix for loading projects from different urls.

nkenneweg committed:
added disabled projects.

Commit:
bug fix: removing a user from a group should also delete project memberships if the user has no own roles left in the project.

apfelfabrik committed:
Merge pull request #47 from opf/fix/member_role_change

bug fix: removing a user from a group should also delete project members...

apfelfabrik committed:
Merge pull request #46 from opf/fix/select2-projectTreeMissing

Fix/select2 project tree missing

apfelfabrik committed:
Speeds up level_list view of projects.

projects.children fetches those children from the DB. this caused the
retrieval of ~2000 projects to take upward of 10 seconds. Asking for
'leaf?' operates on nested set lft and rgt and does not require a
database query.

NobodysNightmare committed:
Merge pull request #49 from opf/fix/slow-level-list

Speeds up level_list view of projects.

ulferts committed:
adapts calling render_details to changes signature

The signature of the render_detail method on journals is changed from (detail, no_html = true) to (detail, :options => { :no_html => false }).

ulferts committed:
bumps rails version to 2.3.18

hschink committed:
Merge pull request #53 from opf/rails/2.3.18

bumps rails version to 2.3.18

nkenneweg committed:
bugfix to re-enable clicking "view all projects":

* adds missing open/close events to top menu
* removes select2's click filter as it masks our menu

Commit:
fix db migrate #1164

apfelfabrik committed:
Merge pull request #50 from opf/fix/pdf_export_journal_details

adapts calling render_details to changes signature

apfelfabrik committed:
Merge pull request #44 from opf/fix/merge_wiki_journal_changes

no longer serializes changes as they are declared to be serialized

GarstgerUnhold committed:
Merge pull request #41 from opf/fix/groups_controller_eager_loading

adds some eager loading to the groups_controller

apfelfabrik committed:
Improves check for api requests.

Timelines introduces a case where the json format is negotiated
but not represented in the format parameter via routes (i.e., the
extension ".json"). This change adds support for Accept-header content
negotiation, which results in the format field of the request. For
consistency, xml was also added.

ulferts committed:
Merge pull request #86 from opf/feature/better-api-request-check

Improves check for api requests.

Commit:
Merge pull request #61 from opf/fix/select2-all-projects-squashed

bugfix to re-enable clicking "view all projects":

nkenneweg committed:
if we are loged out and do ajax, we can redirect to login page.

Commit:
Merge pull request #109 from opf/feature/ajax_session_loss

if we are loged out and do ajax, we can redirect to login page.

ulferts committed:
Merge pull request #132 from marutosi/fix-db-migrate

fix db migrate down update_journals_for_acts_as_journalized

mfrister committed:
Fix XSS in select2 formatter

mfrister committed:
Revert "if we are loged out and do ajax, we can redirect to login page."

This change results in a JavaScript dialog, which breaks cucumber tests.

This reverts commit b3262028dbe96136cc4def5f0d839cddbd925e10.

Commit:
Merge pull request #198 from opf/feature/fix_select2_xss

Fix XSS in select2 formatter

Commit:
add newline at end of lib/tasks/extract_fixtures.rake

ulferts committed:
Merge pull request #245 from marutosi/finn-eol

add newline at end of lib/tasks/extract_fixtures.rake

Commit:
fix indent of app/models/message.rb

Commit:
Merge pull request #285 from marutosi/finn-fix-indent-message

fix indent of app/models/message.rb

Commit:
add newline at end of public/422.html

Commit:
Merge pull request #320 from marutosi/finn-eof

add newline at end of public/422.html

Commit:
fix missing "_test" of test/functional/ldap_auth_sources_controller.rb

ulferts committed:
Merge pull request #380 from marutosi/missing_test_ldap_auth_sources_controller_finn

Missing test ldap auth sources controller

apfelfabrik committed:
Removes comma

According to a bug report by a user, this caused problems in IE 8/9

nkenneweg committed:
Merge pull request #516 from opf/feature/fix-comma

Removes comma

tilt committed:
Split query#available_filter_options and move into mix in

tilt committed:
Rename query#available_filters

tilt committed:
Introduce query filter objects (WIP)

tilt committed:
Add query filter validations, validate values of filters of integer type

tilt committed:
Validate presence of query filter field, remove obsolete attribute accessor

tilt committed:
Forward filter error messages to query

tilt committed:
Add presence of query filter field validation spec

tilt committed:
Avoid duplicate work package query filters

tilt committed:
Fix work package steps with regard to query filters

tilt committed:
Fix query unit tests

tilt commented:
- Query refactoring: WIP
- Integer validations: implemented

Code climate comparison at https://codeclimate.com/repos/52330269f3ea006d790604fa/compare/feature%2Fquery-refactoring-and-validations-2395

AvailableFilterOptions has been refactored out of query model and only slightly tidied.

tilt committed:
Fix error message assertion when query filter cannot be blank

tilt commented:
TODO: for backwards compatibility of serialized filters, the keys have to be stringified...

hschink commented:
- [x] Move files from `lib/query` to `app/models/query`
- [x] Fix plug-in dependencies
- [x] **BUG**: Status is set in any case

tilt committed:
Move query filter-related code to models/query because it existed first

tilt committed:
Move filter types by fields concerning work packages to corresponding filter subclass

tilt committed:
Stringify keys of serialized filter hashes


This ensures backwards code compatibility. Can be reverted for aesthetic reasons later on.

Commit:
Merge branch 'feature/rails3' into 'master'

tilt committed:
Only initialize queries with status filters if no filters have been selected by user

tilt committed:
Validate presence of query filters

tilt committed:
Ensure valid setup of queries from factory

apfelfabrik committed:
Removes alerts for missing permissions.

These appear to not be thought through w/ anonymous users. Those have
little rights, and can see no timeline w/o reportings rights. This
leaves the X-Reason header in the server side, since I'm not sure that
there isn't an API client depending on it.

apfelfabrik committed:
Changelog.

Commit:
let travis build the new dev branch

hschink committed:
Merge pull request #691 from opf/feature/remove-login-alert-3061

tilt committed:
Fix work packages controller spec after changing the query default filter setup

tilt committed:
Fix unit tests after changing the default query filter setup

tilt committed:
Fix query integration test after changing the default query filter setup

Commit:
Removes deprecated cuke

Removes cuke that checks conditions that do not hold since the merge
of OpenProject ticket #3061.

tilt committed:
Add available filters alias method to queries


Currently there are no queries / filters other than those for work packages

tilt committed:
Fix some application css
